docs/advanced/document-store/index.md

```diff
@@ -201,3 +201,65 @@ da[0, 'text'] = 'hello'
 Obviously, a DocumentArray with on-disk storage is slower than in-memory DocumentArray. However, if you choose to use on-disk storage, then often your concern of persistence overwhelms the concern of efficiency.
  
 Slowness can affect all functions of DocumentArray. On the bright side, they may not be that severe as you would expect. Modern database are highly optimized. Moreover, some database provides faster method for resolving certain queries, e.g. nearest-neighbour queries. We are actively and continuously improving DocArray to better leverage those features. 
+
+
+## Benchmarking
+The script `scripts/benchmarking.py` benchmarks DocArray's supported Document Stores in 6 different operations:
+* Create (indexing Documents)
+* Read
+* Update
+* Delete
+* Find Document by vector (Nearest Neighbor Search or Approximate Nearest Neighbor Search depending on the `backend`)
+* Find Document by condition (apply filter)
+
+Since most of these Document Stores use their implementation of the HNSW Approximate Nearest Neighbor Search algorithm, 
+with various default HNSW parameters, we conducted 2 benchmarking experiments for the `Find By Vector` operation:
+* Set up the Document Stores with the same HNSW parameters
+* Set up the Document Stores with their default HNSW parameters at the time the benchmarking experiment was conducted.
+
+Furthermore, we provide the `Recall At K` value, considering the exhaustive search as the ground truth. This allows 
+you to also take into consideration the quality, not just the speed.
+
+```{important}
+The Sqlite and the in-memory Document Stores ** do not implement ** approximate nearest neighbor search and offer 
+exhaustive search instead. That's why, they give the maximum quality but are the slowest when it comes to search nearest 
+neighbors.
+```
+
+The results were conducted on a 4.5 Ghz Quad-Core i7700k processor with Python 3.8.5 and using the official docker 
+images of the storage backends. The docker images were allocated 40 GB of RAM.
+
+The benchmarking experiments used the following parameters:
+* Number of indexed Documents: 1M
+* Number of query Documents: 1
+* Embedding dimension: 128
+* Number of Documents (K) to be retrieved for `Find Document by vector`: 10
+
+For the first experiment, we fixed the following HNSW parameters for the Document Stores that support ANN:
+* `ef` (construction): 100
+* `ef` (search): 100
+* `m` (max connections): 16
+
+````{tab} #1: Same HNSW parameters
```

Comment:
Move tables to the top, right after 2~3 intro sentences.

Reply:
done